src/dataService/savedItemsService.ts

```diff
@@ -326,9 +326,16 @@ export class SavedItemDataService {
    * which status the item is assigned when moved from the deleted state.
    * @param itemId
    */
-  public async updateSavedItemUnDelete(itemId): Promise<SavedItem> {
+  public async updateSavedItemUnDelete(
+    itemId: string,
+    updatedAt?: Date
+  ): Promise<SavedItem | null> {
+    const timestamp = updatedAt ?? SavedItemDataService.formatDate(new Date());
     const query: any = await this.getSavedItemTimeRead(itemId);
-
+    // Does not exist or was hard-deleted
```

Comment:
This would previously throw an unintuitive server error (would fail on line 341, attempting to access key on undefined). This doesn't change the end state behavior but results in a better response. The null gets caught by the calling function and is converted to NOT FOUND in business logic layer.